src/runtime/middleware/auth.ts

```diff
@@ -105,6 +105,11 @@ export default defineNuxtRouteMiddleware((to) => {
     return navigateTo(metaAuth.navigateUnauthenticatedTo)
   }
   else {
-    return navigateTo(authConfig.provider.pages.login)
+    return navigateTo({
+      path: authConfig.provider.pages.login,
+      query: {
+        redirect: to.fullPath
```

Comment:
Only add this if `globalAppMiddleware.addDefaultCallbackUrl` is `true` (Note: it can also be a string, if this is the case, add that string as the redirect` path)

Reply:
Thanks @zoey-kaiser ! I have updated as you suggested.